fix: only store one node info and update that

dknopik opened the pull request:
## Issue Addressed

We recently introduced a regression where we fail all handshakes. The root cause for this is that we started storing the `NodeInfo` in a new place but did not adjust all places to use this.

## Proposed Changes

- Remove the field `Network::node_info`
- Add accessors for the node metadata
- Adjust old references of `Network::node_info` to use new accessors instead

dknopik committed:
fix: only store one node info and update that

diegomrsantos commented on anchor/network/src/network.rs:
```diff
@@ -571,7 +563,7 @@ impl<R: MessageReceiver> Network<R> {
         their_metadata: &NodeMetadata,
     ) {
-        if let Some(our_metadata) = &self.node_info.metadata {
+        if let Some(our_metadata) = self.node_metadata() {
```
Should this function be moved to the handshake module?

dknopik replied:
there is a `node_metadata` function on the handshake module, I just added another one in `network.rs` as a shortcut, calling the one in the handshake module.

diegomrsantos replied:
I meant `record_handshake_subnet_match_metrics`

diegomrsantos commented:
@claude review this PR

diegomrsantos commented:
@claude suggest how a test like the one described below could be implemented:

> While the existing handshake tests in handshake/mod.rs (lines 347-440) provide good coverage of the protocol, consider adding an integration test that:
Subscribes to a subnet
Verifies the metadata is correctly updated via the accessor methods
Confirms the updated metadata is sent during subsequent handshakes
This would verify the end-to-end flow of the regression fix.

dknopik committed:
move fn

diegomrsantos committed:
refactor(network): consolidate handshake metrics and logging in handshake module

Moved handshake completion logging and subnet metrics recording into the
handshake module's verify_and_emit_event method, eliminating the need for
network.rs to call back into the handshake module.

Changes:
- Removed public record_handshake_subnet_match_metrics method
- Consolidated logging and metrics recording in verify_and_emit_event
- Eliminated duplicate subnet matching calculations
- Removed callback from network.rs handle_handshake_result

This improves separation of concerns, with the handshake module now fully
owning its internal metrics and logging, while network.rs focuses on
coordination between components.

diegomrsantos committed:
refactor(network): add handshake() helper for consistent behavior access

Added handshake() helper method to match the existing pattern for accessing
behavior components (peer_manager, gossipsub, discovery). Removed the
node_metadata() and node_metadata_mut() helper methods which were reaching
two levels deep into behavior internals.

Changes:
- Added handshake() method returning &mut handshake::Behaviour
- Removed node_metadata() helper (unused)
- Removed node_metadata_mut() helper (single use)
- Updated usage to self.handshake().node_metadata_mut()
- Removed unused node_info::NodeMetadata import

This maintains a consistent single-level indirection pattern where helpers
provide access to behaviors, not their internal state.
